bundle/src/main/java/com/adobe/acs/commons/rewriter/impl/ContentVariableTransformer.java

```diff
@@ -0,0 +1,116 @@
+        this.aggregatorService = propertyAggregatorService;
+    }
+
+    @Override
+    public void init(ProcessingContext processingContext, ProcessingComponentConfiguration processingComponentConfiguration) throws IOException {
+        SlingHttpServletRequest request = processingContext.getRequest();
+
+        contentVariableReplacements = aggregatorService.getProperties(request);
+    }
+
+    public void startElement(String uri, String localName, String quaName, Attributes atts) throws SAXException {
+        if (shouldRun() && localName.equals("a")) {
+            AttributesImpl newAttrs = new AttributesImpl(atts);
+            for (int i = 0; i < newAttrs.getLength(); i++) {
+                String currentAttribute = newAttrs.getValue(i);
+                if (ContentVariableReplacementUtil.hasPlaceholder(currentAttribute)) {
+
+                    // Get the current placeholder in the string
+                    String placeholder = ContentVariableReplacementUtil.getPlaceholder(currentAttribute);
+
+                    // Transform it to the key in the property map
+                    String key = ContentVariableReplacementUtil.getKey(placeholder);
+
+                    // If the placeholder key is in the map then replace it
+                    if (contentVariableReplacements.containsKey(key)) {
+                        String replaceValue = (String) contentVariableReplacements.get(key);
+                        newAttrs.setValue(i, currentAttribute.replace(placeholder, replaceValue));
+                    }
+                }
+            }
+            getContentHandler().startElement(uri, localName, quaName, newAttrs);
+        } else {
+            getContentHandler().startElement(uri, localName, quaName, atts);
+        }
+    }
+
+    public void characters(char[] ch, int start, int length) throws SAXException {
+        String currentString = new String(ch);
+        int chunkLength = length;
+
+        if (shouldRun()) {
+            String currentChunk = currentString.substring(start, start + length);
+
+            // Get the current placeholders in the string
+            final List<String> placeholders = ContentVariableReplacementUtil.getPlaceholders(currentChunk);
+
+            for (String placeholder : placeholders) {
+                // Transform it to the key in the property map
+                final String key = ContentVariableReplacementUtil.getKey(placeholder);
+
+                // If the placeholder key is in the map then replace it
+                if (contentVariableReplacements.containsKey(key)) {
+                    final String placeholderReplacement = String.valueOf(contentVariableReplacements.get(key));
+                    final String replacedChunk = currentChunk.replace(placeholder, placeholderReplacement);
+                    chunkLength = replacedChunk.length();
+                    currentString = currentString.replace(currentChunk, replacedChunk);
```

Comment:
Could this accidentally replace multiple instances of `currentChunk` and could that result in unexpected behavior?

Reply:
I haven't experienced that issue, but i think it's possible depending on how the chunks for any given page being piped through this are broken down.